src/core/annotation.js

```diff
@@ -980,6 +991,31 @@ class ButtonWidgetAnnotation extends WidgetAnnotation {
     }
   }
 
+  getOperatorList(evaluator, task, renderForms, annotationStorage) {
+    if (annotationStorage) {
+      const value = annotationStorage[this.data.id] || false;
+      if (value && this.checkedAppearance) {
+        const savedAppearance = this.appearance;
+        this.appearance = this.checkedAppearance;
+        const operatorList = super.getOperatorList(
+          evaluator,
+          task,
+          renderForms,
+          annotationStorage
+        );
+        this.appearance = savedAppearance;
+        return operatorList;
+      }
+      return Promise.resolve(new OperatorList());
```

Comment:
Shouldn't we treat this case just like the caser where we have no `annotationStorage`, i.e., let this fall through and let the parent class handle it? I would think we only want to special-case the case where we have a value and a checked appearance; any other case would need the default parent `getOperatorList` call.

Reply:
If we've a checked box (in the base pdf) and we uncheck it then falling through implies that we'll print a checked box.
Anyway in reading the spec, we can have a "Off" appearance so I need to handle that case.